docs/reference/migration/migrate_7_0/migrate_to_java_time.asciidoc

```diff
@@ -0,0 +1,350 @@
+         }
+      }
+    }
+  }
+}
+--------------------------------------------------
+
+
+* Create my_index_2 with mappings changed with the format set to `8uuuu/MM/dd HH:mm:ss||uuuu/MM/dd||epoch_millis`
+
+[source,console]
+--------------------------------------------------
+PUT my_index_2
+{
+  "mappings": {
+    "properties": {
+      "datetime": {
+        "type": "date",
+        "format": "8uuuu/MM/dd HH:mm:ss||uuuu/MM/dd||epoch_millis"
+      }
+    }
+  }
+}
+--------------------------------------------------
+// TEST[continued]
+
+* Reindex the deprecated format index to new_index_1
+
+[source,console]
+--------------------------------------------------
+POST _reindex
+{
+  "source": {
+    "index": "my_index_1"
+  },
+  "dest": {
+    "index": "my_index_2"
+  }
+}
+--------------------------------------------------
+// TEST[continued]
+
+* If you were using aliases, update them to a new index
+
+[source,console]
+--------------------------------------------------
+POST /_aliases
+{
+    "actions" : [
+        { "remove" : { "index" : "my_index_1", "alias" : "my_index" } },
+        { "add" : { "index" : "my_index_2", "alias" : "my_index" } }
+    ]
+}
+--------------------------------------------------
+// TEST[continued]
+
+===== Update before upgrading to ES7.
+If your pipelines were using a joda style patterns, they also have to be updated. There is no need to create a new pipeline.
+Just update the already existing one.
+
+[source,console]
+--------------------------------------------------
+PUT _ingest/pipeline/mypipeline
+{
+  "description": "Pipeline for routing data to specific index",
+  "processors": [
+    {
+      "date": {
+        "field": "createdTime",
+        "formats": [
+         "8uuuu-w"
+        ]
+      },
+      "date_index_name": {
+        "field": "@timestamp",
+        "date_rounding": "d",
+        "index_name_prefix": "x-",
+        "index_name_format": "8uuuu-w"
+      }
+    }
+  ]
+}
+--------------------------------------------------
+
+
+* upgrade template
+If your template was using joda date pattern it also should be updated before upgrading to ES7.
+
+[source,console]
+--------------------------------------------------
+PUT _template/template_1
+{
+  "index_patterns": [
+    "te*",
+    "bar*"
+  ],
+  "settings": {
+    "number_of_shards": 1
+  },
+  "mappings": {
+    "_source": {
+      "enabled": false
+    },
+    "properties": {
+      "host_name": {
+        "type": "keyword"
+      },
+      "created_at": {
+        "type": "date",
+        "format": "8EEE MMM dd HH:mm:ss Z yyyy"
+      }
+    }
+  }
+}
+--------------------------------------------------
+
+
+[source,console]
+--------------------------------------------------
+DELETE /_template/template_1
+--------------------------------------------------
+// TEST[continued]
+* Upgrade to 7.x
+
+===== External templates
+Revisit other templates from elastic stack where you used a custom date pattern.
```

Comment:
```suggestion
[[java-time-migration-update-external-tools-templates]]
===== Update external tools and templates
When updating custom date formats, ensure you also update any templates or tools
used _outside_ of {es}.

This can include tools such as {beats-ref}/getting-started.html[{beats}] or
{logstash-ref}/index.html[Logstash].
```